### PR TITLE
Delete state from project when changing dataset

### PR DESCRIPTION
**Pull request by J535D165**
Not tested yet.

### CODE REVIEW

File: asreview/state/utils.py [hunk 1]
```diff
@@ -200,6 +200,39 @@ def open_state(working_dir, review_id=None, read_only=True):
...
+    with open(project_file_path, "r") as f_read:
+        project_dict = json.load(f_read)
+
+    # add path to dict (overwrite if already exists)
+    reset_config = {
+        'projectInitReady': False,
+        'reviewFinished': False,
+        'reviews': [],
+        'feature_matrices': []
+    }
```
Comment (terrymyc) [inline asreview/state/utils.py:229]: The key `projectHasDataset` has been deprecated? Now when the value of `reviews` is an empty array, it means no dataset?

**Review by terrymyc — APPROVED** [review]

File: asreview/state/utils.py (hunk 1 above)
Comment (J535D165) [reply asreview/state/utils.py:229]: I don't think so. `projectHasDataset` is generated on the fly in 

https://github.com/asreview/asreview/blob/6af53348b67c16b0c6579a4871ff0b6137e351b2/asreview/webapp/api.py#L198-L203

`reviews` doesn't contain datasets. We decided to put only one dataset in a project, but with the option to have multiple reviews (useful for simulation in the future).

**Comment by J535D165** [comment]
Not clear to me why the tests fail :(

**Comment by J535D165** [comment]
Fixed the tests. There was a boolean wrong.

**Comment by terrymyc** [comment]
Close #840

**Pull request merged**